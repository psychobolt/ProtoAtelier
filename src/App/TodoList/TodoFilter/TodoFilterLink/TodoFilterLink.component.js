// @flow
import React from 'react';

const TodoFilterLink = (
  { active, goto, children }: {active?: boolean, goto: Function, children: any },
) => (
  <x-tab
    onClick={goto}
    selected={active || null}
  >
    <x-label>{children}</x-label>
  </x-tab>
);

TodoFilterLink.defaultProps = {
  active: false,
};

export default TodoFilterLink;
