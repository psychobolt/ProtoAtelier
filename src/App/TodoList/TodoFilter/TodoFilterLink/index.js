export { default } from './TodoFilterLink.container';
