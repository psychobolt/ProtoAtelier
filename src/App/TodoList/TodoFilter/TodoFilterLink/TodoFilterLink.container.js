import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import TodoFilterLink from './TodoFilterLink.component';
import Filters from '../TodoFilter.filters';

const mapParamToFilter = param => param || Filters.ALL;

export const ROOT = '/';

const mapStateToProps = (state, ownProps) => ({
  active: ownProps.filter === mapParamToFilter(ownProps.match.params.filter),
  goto: () => {
    ownProps.history.push(ownProps.filter === Filters.ALL ? ROOT : ownProps.filter);
  },
});

export default withRouter(connect(mapStateToProps)(TodoFilterLink));
