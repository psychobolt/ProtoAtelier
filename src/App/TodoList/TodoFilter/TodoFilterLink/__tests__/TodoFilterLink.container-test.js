import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import renderTest from 'Framework/ReactRouterRenderTest';
import TodoFilterLink, { ROOT } from '../TodoFilterLink.container';
import Filters from '../../TodoFilter.filters';

const mockStore = configureMockStore([]);

describe('container <TodoFilterLink />', () => {
  it('should render correctly without crashing', () => {
    shallow(<TodoFilterLink filter={Filters.ALL}>{Filters.ALL}</TodoFilterLink>, {
      context: { store: mockStore({}) },
    });
  });

  it('should go to / when ALL filter is clicked', () => {
    const props = {
      store: mockStore({}),
      filter: Filters.ALL,
    };
    const pathname = `/${Filters.ACTIVE}`;
    const wrapper = renderTest(<TodoFilterLink {...props}>{props.filter}</TodoFilterLink>, {
      initialEntries: [`/${Filters.ACTIVE}`],
      initialIndex: 0,
      steps: [
        ({ history }) => {
          expect(history.location.pathname).toBe(pathname);
        },
        ({ history }) => {
          expect(history.location.pathname).toBe(ROOT);
        },
      ],
    });
    wrapper.find('x-tab').simulate('click');
  });

  it('should go to /active when ACTIVE filter is clicked', () => {
    const props = {
      store: mockStore({}),
      filter: Filters.ACTIVE,
    };
    const pathname = `/${props.filter}`;
    const wrapper = renderTest(<TodoFilterLink {...props}>{props.filter}</TodoFilterLink>, {
      initialEntries: [ROOT],
      initialIndex: 0,
      steps: [
        ({ history }) => {
          expect(history.location.pathname).toBe(ROOT);
        },
        ({ history }) => {
          expect(history.location.pathname).toBe(pathname);
        },
      ],
    });
    wrapper.find('x-tab').simulate('click');
  });
});
