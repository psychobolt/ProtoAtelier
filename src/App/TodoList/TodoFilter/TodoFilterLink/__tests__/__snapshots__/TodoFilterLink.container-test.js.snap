// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <TodoFilterLink /> should render correctly 1`] = `
<Route
  render={[Function]}
/>
`;
